applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h

```diff
@@ -0,0 +1,376 @@
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Sergio Jimenez/Alejandro Cornejo/Lucia Barbu
+//  Collaborator:    
+//
+
+#if !defined(KRATOS_GENERIC_SMALL_STRAIN_HIGH_CYCLE_FATIGUE_LAW_H_INCLUDED)
+#define KRATOS_GENERIC_SMALL_STRAIN_HIGH_CYCLE_FATIGUE_LAW_H_INCLUDED
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_constitutive/generic_small_strain_isotropic_damage.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+// The size type definition
+typedef std::size_t SizeType;
+    
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+/**
+ * @class GenericSmallStrainHighCycleFatigueLaw
+ * @ingroup StructuralMechanicsApplication
+ * @brief This class is the base class which defines the constitutive law used for high cycle fatigue (HCF) in small deformation
+ * @details This class uses the GenericSmallStrainIsotropicDamage class once the load is applied Nf cycles. The code has been written following the approach proposed by S. Oller et al. in A continuum mechanics model for mechanical fatigue analysis (2005)
+ * @tparam TConstLawIntegratorType The constitutive law integrator considered
+ * @author Sergio Jiménez/Alejandro Cornejo/Lucia Barbu
+ */
+template <class TConstLawIntegratorType>
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) GenericSmallStrainHighCycleFatigueLaw
+    : public GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// The define the working dimension size, already defined in the integrator
+    static constexpr SizeType Dimension = TConstLawIntegratorType::Dimension;
+
+    /// The define the Voigt size, already defined in the  integrator
+    static constexpr SizeType VoigtSize = TConstLawIntegratorType::VoigtSize;
+    
+    /// Counted pointer of GenericYieldSurface
+    KRATOS_CLASS_POINTER_DEFINITION(GenericSmallStrainHighCycleFatigueLaw);
+
+    /// The node definition
+    typedef Node<3> NodeType;
+    
+    /// The geometry definition
+    typedef Geometry<NodeType> GeometryType;
+    
+    /// Definition of the machine precision tolerance
+    static constexpr double tolerance = std::numeric_limits<double>::epsilon();
+
+    /// Definition of the base class
+    //typedef typename GenericSmallStrainIsotropicDamage<TConstLawIntegratorType> BaseType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+    * Default constructor.
+    */
+    GenericSmallStrainHighCycleFatigueLaw()
+    {
+    }
+
+    /**
+    * Clone.
+    */
+    ConstitutiveLaw::Pointer Clone() const override
+    {
+        return Kratos::make_shared<GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>>(*this);
+    }
+
+    /**
+    * Copy constructor.
+    */
```

Comment:
assignment operator should go with copy constructor.

Reply:
done